### PR TITLE
FreeBSD service scripts

### PR DESCRIPTION
**Pull request by mattias-p**
This purpose of this PR is to add separate FreeBSD service scripts for the RPCAPI and TestAgent daemons. Having two single-daemon services instead of one dual-daemon script has two advantages:
* Users get more control.
* The implementation doesn't have to consider complex failure modes where one operation succeeds and the other one fails.

Additional drive-by fixes:
* The paths where log and pid files are written (from $HOME to /var)
* A zonemaster user is created and used (like on CentOS)
* File permissions are tweaked to be more restrictive (and reasonable)

### CODE REVIEW

File: docs/Installation.md [hunk 1]
```diff
@@ -514,14 +518,15 @@ psql -U pgsql -f ./initial-postgres.sql template1
...
->
+Install service scripts and start services:
 
 ```sh
-starman --error-log="$HOME/logs/error.log" --pid-file="$HOME/logs/starman.pid" --listen=127.0.0.1:5000 --daemonize /usr/local/bin/zonemaster_backend_rpcapi.psgi 
-zonemaster_backend_testagent start
+install -m 775 -g zonemaster -d /var/log/zonemaster
+install -m 775 -g zonemaster -d /var/run/zonemaster
+install -m 755 ./zm_rpcapi /usr/local/etc/rc.d/
+install -m 755 ./zm_testagent /usr/local/etc/rc.d/
```
Comment (matsduf) [inline docs/Installation.md]: For the file names, see MANIFEST above.

File: docs/Installation.md [hunk 2]
```diff
@@ -514,14 +518,15 @@ psql -U pgsql -f ./initial-postgres.sql template1
...
+echo 'zm_rpcapi_enable="YES"' | tee -a /etc/rc.conf
+echo 'zm_testagent_enable="YES"' | tee -a /etc/rc.conf
 ```
```
Comment (matsduf) [inline docs/Installation.md]: I suggest the following lines instead as it is usually written i FreeBDS instructions (e.g. FreeBSD Handbook):

Add the following lines to /etc/rc.conf:
```
zm_rpcapi_enable="YES"
zm_testagent_enable="YES"
```
Then start the daemons:
```
service  zm_rpcapi start
service zm_testagent start
```

File: share/zm_rpcapi [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# PROVIDE: zm_rpcapi
+# REQUIRE: NETWORKING mysql postgresql
```
Comment (matsduf) [inline share/zm_rpcapi]: What happens if not both of the database engines are installed?

File: share/zm_testagent [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+# PROVIDE: zm_testagent
+# REQUIRE: mysql postgresql
```
Comment (matsduf) [inline share/zm_testagent]: What happens if not both of the database engines are installed?

File: docs/Installation.md [hunk 5]
```diff
@@ -425,6 +425,12 @@ cpan -i Zonemaster::Backend
 > The command above might try to install "DBD::Pg" and "DBD::mysql".
 > You can ignore if it fails. The relevant libraries are installed further down in these instructions.
 
+Add `zonemaster` user and group:
+```sh
+pw groupadd zonemaster
+pw useradd zonemaster -g zonemaster -s /sbin/nologin -d /nonexistent -c "Zonemaster daemon user"
```
[the remark this answers is not in the document]
Comment (mattias-p) [reply docs/Installation.md]: I actually tried that and failed.

File: share/zm_rpcapi (hunk 3 above)
Comment (mattias-p) [reply share/zm_rpcapi]: Regarding the meaning of REQUIRE I quote https://www.freebsd.org/cgi/man.cgi?rcorder(8)#end:

> The "REQUIRE" keyword is misleading: It doesn't describe which daemons have to be running before a script will be started. It describes which scripts must be placed before it in the dependency ordering. For example, if your script has a "REQUIRE" on "named", it means the script must be placed after the "named" script in the dependency ordering, not necessarily that it requires named(8) to be started or enabled.

My idea for dealing with unavailable database engines is described in #369. I believe, thought I haven't thoroughly tested it, that the service script part of #369 is implemented automagically in these scripts by virtue of the rc.d framework.

File: share/zm_testagent (hunk 4 above)
Comment (mattias-p) [reply share/zm_testagent]: Ditto.

File: docs/Installation.md (hunk 2 above)
Comment (mattias-p) [reply docs/Installation.md]: The `echo | tee -a` lines have the advantage that they are cut-and-pasteable, and I reckon we'll be adding those lines a lot.

Comment (matsduf) [reply docs/Installation.md]: The `echo | tee -a` line assumes that rc.conf ends with a line ending. It also adds that line even if not needed (already there). Here is a new version:

```
Run the following command turn the daemons on in /etc/rc.conf:

sysrc zm_rpcapi_enable="YES"
sysrc zm_testagent_enable="YES"
```

Comment (mattias-p) [reply docs/Installation.md]: Neat! That's much better.

**Comment by mattias-p** [comment]
@matsduf Can you take another look at this?

File: docs/Installation.md [hunk 6]
```diff
@@ -514,14 +518,22 @@ psql -U pgsql -f ./initial-postgres.sql template1
...
->
+Install service scripts:
+
+```sh
+install -m 775 -g zonemaster -d /var/log/zonemaster
+install -m 775 -g zonemaster -d /var/run/zonemaster
+install -m 755 ./zm_rpcapi-bsd /usr/local/etc/rc.d/zm_rpcapi
+install -m 755 ./zm_testagent-bsd /usr/local/etc/rc.d/zm_testagent
+echo 'zm_rpcapi_enable="YES"' | tee -a /etc/rc.conf
+echo 'zm_testagent_enable="YES"' | tee -a /etc/rc.conf
```
Comment (matsduf) [inline docs/Installation.md]: I propose the following instead:

sysrc zm_rpcapi_enable="YES"
sysrc zm_testagent_enable="YES"

File: docs/Installation.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/Installation.md]: Ok.

File: docs/Installation.md (hunk 6 above)
Comment (mattias-p) [reply docs/Installation.md]: Sorry, I forgot that.. Addressed it now.

**Pull request merged**